presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -476,4 +523,35 @@ private static void checkInputs(
     {
         return scala.reflect.ClassTag$.MODULE$.apply(clazz);
     }
+
+    private static class SparkPartition
+            implements Comparable<SparkPartition>
+    {
+        private final int partitionId;
+        private long splitsSizeInBytes;
+
+        public SparkPartition(int partitionId)
+        {
+            this.partitionId = partitionId;
+        }
+
+        @Override
+        public int compareTo(SparkPartition o)
+        {
+            return splitsSizeInBytes == o.splitsSizeInBytes ?
+                    0 :
+                    splitsSizeInBytes < o.splitsSizeInBytes ? -1 : 1;
+        }
+
+        public int getPartitionId()
+        {
+            return partitionId;
+        }
+
+        public void assignSplit(ScheduledSplit scheduledSplit)
+        {
+            OptionalLong splitSizeInBytes = scheduledSplit.getSplit().getConnectorSplit().getSplitSizeInBytes();
+            splitsSizeInBytes += splitSizeInBytes.orElseThrow(() -> new IllegalArgumentException("splitSizeInBytes is not present"));
```

Comment:
This breakes the non auto tune path. It may start throwing if the split size is not available

Reply:
Good catch ! Is it possible that some splits have size available and some don't, for this case, a combination of even distribution + random distribution will be used.